src/server/rollback_sqlite3.cpp

```diff
@@ -0,0 +1,617 @@
+		row.nodeMeta  = 0;
+
+		if (row.type == RollbackAction::TYPE_MODIFY_INVENTORY_STACK) {
+			text = sqlite3_column_text(stmt, 3);
+			size = sqlite3_column_bytes(stmt, 3);
+			row.list        = std::string(reinterpret_cast<const char *>(text), size);
+			row.index       = sqlite3_column_int(stmt, 4);
+			row.add         = sqlite3_column_int(stmt, 5);
+			row.stack.id    = sqlite3_column_int(stmt, 6);
+			row.stack.count = sqlite3_column_int(stmt, 7);
+			row.nodeMeta    = sqlite3_column_int(stmt, 8);
+		}
+
+		if (row.type == RollbackAction::TYPE_SET_NODE || row.nodeMeta) {
+			row.x = sqlite3_column_int(stmt,  9);
+			row.y = sqlite3_column_int(stmt, 10);
+			row.z = sqlite3_column_int(stmt, 11);
+		}
+
+		if (row.type == RollbackAction::TYPE_SET_NODE) {
+			row.oldNode   = sqlite3_column_int(stmt, 12);
+			row.oldParam1 = sqlite3_column_int(stmt, 13);
+			row.oldParam2 = sqlite3_column_int(stmt, 14);
+			text = sqlite3_column_text(stmt, 15);
+			size = sqlite3_column_bytes(stmt, 15);
+			row.oldMeta   = std::string(reinterpret_cast<const char *>(text), size);
+			row.newNode   = sqlite3_column_int(stmt, 16);
+			row.newParam1 = sqlite3_column_int(stmt, 17);
+			row.newParam2 = sqlite3_column_int(stmt, 18);
+			text = sqlite3_column_text(stmt, 19);
+			size = sqlite3_column_bytes(stmt, 19);
+			row.newMeta   = std::string(reinterpret_cast<const char *>(text), size);
+			row.guessed   = sqlite3_column_int(stmt, 20);
+		}
+
+		if (row.nodeMeta) {
+			row.location = "nodemeta:";
+			row.location += itos(row.x);
+			row.location += ',';
+			row.location += itos(row.y);
+			row.location += ',';
+			row.location += itos(row.z);
+		} else {
+			row.location = getActorName(row.actor);
+		}
+
+		rows.push_back(row);
+	}
+
+	SQLOK(sqlite3_reset(stmt));
+	return rows;
+}
+
+ActionRow RollbackManagerSQLite3::actionRowFromRollbackAction(const RollbackAction &action)
+{
+	ActionRow row;
+	row.actor     = getActorId(action.actor);
+	row.timestamp = action.unix_time;
+	row.type      = action.type;
+
+	if (row.type == RollbackAction::TYPE_MODIFY_INVENTORY_STACK) {
+		row.location = action.inventory_location;
+		row.list     = action.inventory_list;
+		row.index    = action.inventory_index;
+		row.add      = action.inventory_add;
+		row.stack    = action.inventory_stack;
+		row.stack.id = getNodeId(row.stack.name);
+	} else {
+		row.x         = action.p.X;
+		row.y         = action.p.Y;
+		row.z         = action.p.Z;
+		row.oldNode   = getNodeId(action.n_old.name);
+		row.oldParam1 = action.n_old.param1;
+		row.oldParam2 = action.n_old.param2;
+		row.oldMeta   = action.n_old.meta;
+		row.newNode   = getNodeId(action.n_new.name);
+		row.newParam1 = action.n_new.param1;
+		row.newParam2 = action.n_new.param2;
+		row.newMeta   = action.n_new.meta;
+		row.guessed   = action.actor_is_guess;
+	}
+
+	return row;
+}
+
+std::list<RollbackAction> RollbackManagerSQLite3::rollbackActionsFromActionRows(const std::list<ActionRow> &rows)
```

Comment:
> Try to keep lines under 95 characters

https://docs.luanti.org/for-engine-devs/code-style-guidelines/

Multiple occurrences. Perhaps it might make sense to use a shorter class name, such as `RollbackMgr`? (This abbreviation is already used in other classes).

Reply:
RollbackMgr will replace RollbackManager in the Rollback code. Thanks.